src/coreclr/src/jit/block.h

```diff
@@ -514,16 +514,14 @@ struct BasicBlock : private LIR::Range
     const char* dspToString(int blockNumPadding = 0);
 #endif // DEBUG
 
-    typedef unsigned weight_t; // Type used to hold block and edge weights
-                               // Note that for CLR v2.0 and earlier our
-                               // block weights were stored using unsigned shorts
+    // Type used to hold block and edge weights
+    typedef float weight_t;
```

Comment:
Is it worthwhile to keep `weight_t` as a BasicBlock member, or should we just make it a global type? (Or even its own class?)

Reply:
I'd probably vote to get rid of it entirely and just use `float`. I don't find these sorts of typedefs helpful; most anywhere you work with these values you also need to know the actual underlying type.

In past compilers I've seen people wrap these floats in structs to get stronger type checking but that is a whole lot of uninteresting boilerplate.